swagger.yaml

```diff
@@ -223,52 +206,120 @@ paths:
         type: string
-        description: "The title for the dataset"
-      url:
+        description: "An unique id for a dataset"
+        example: "DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC"
+      theme:
         type: string
-        description: "An url to the dataset"
-      release_date:
+        description: "A theme for a dataset"
+      periodicity:
         type: string
-        description: "The release date of this dataset"
-      edition:
+        description: "The periodicity for a dataset"
+      publisher:
+        description: "The publisher of the dataset"
+        $ref: '#/definitions/Publisher'
+      latest_url:
         type: string
-        description: "The edition of the dataset"
-      version:
+        description: "A link to the latest version and edition of the dataset"
+      next_release:
         type: string
-        description: "The version of the dataset"
+        description: "The next release date for this dataset"
       contact:
-        type: object
+        type: array
         description: "Object containing contact information for this dataset"
-        properties:
-          name:
-            type: string
-            description: "The name of the person"
-          telephone:
-            type: string
-            description: "Telephone number to contact the person"
-          email:
-            type: string
-            description: "An email address to contact the person"
+        items:
+          $ref: '#/definitions/Contact'
+      editions_url:
+        type: string
+        description: "A URL to all editions for a dataset"
+        example: "http://localhost:8080/datasets/DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC/editions"
   Datasets:
     type: object
     properties:
       items:
         type: array
-        description: "An array of Datasets"
+        description: "A list of datasets"
         items:
           $ref: '#/definitions/Dataset'
-      number_of_results:
-        type: integer
-        description: "The number of results available"
+  Edition:
+    type: object
+    properties:
+      id:
+        type: string
+        description: "An unique id for a dataset edtion"
+        example: "2017"
+      versions_url:
+        type: string
+        description: "A URL to all versions for this dataset edition"
+        example: "http://localhost:8080/datasets/DE3BC0B6-D6C4-4E20-917E-95D7EA8C91DC/editions/2017/versions"
+  Editions:
+    type: object
+    properties:
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/Edition'
+  Versions:
+    type: object
+    properties:
+      items:
+        type: array
+        description: "An array of Datasets"
+        items:
+          $ref: '#/definitions/Version'
+  Version:
+    type: object
+    description: "An object containing information about published datasets from the ONS"
+    properties:
+      id:
+        type: string
+        description: "An unique id for this version of a dataset"
+      title:
```

Comment:
Title, description and license I believe will be at the dataset level (sorry, the list I sent you may have been unclear - if you've spoken to Rob about this and he's said this is the right way, that overrides my thought)

Reply:
I spoken to rob about the license property, we agreed it should be on a version level as once a dataset is published under that license(and version of it). It can not be changed. Putting it on the dataset level means any amends to the current license or changing the license would affect older datasets which have been published with a different license.

For the title and description, I think i seen examples where it contains edition details and versions. I validate this with rob

Reply:
I updated the models, so the title and descriptions has moved back into the dataset level